--- origsrc/gnutls-3.1.4/guile/src/Makefile.am	2012-08-11 15:00:04.000000000 +0200
+++ src/gnutls-3.1.4/guile/src/Makefile.am	2012-11-17 16:50:28.544142100 +0100
@@ -32,7 +32,7 @@ BUILT_SOURCES = enum-map.i.c smob-types.
 CLEANFILES = $(BUILT_SOURCES)
 
 guileextension_LTLIBRARIES  = guile-gnutls-v-2.la
-guile_gnutls_v_2_la_LDFLAGS = -module
+guile_gnutls_v_2_la_LDFLAGS = -module -avoid-version -no-undefined
 
 # Linking against GnuTLS.
 GNUTLS_CORE_LIBS  = $(top_builddir)/lib/libgnutls.la
