--- origsrc/gnutls-3.3.27/tests/Makefile.am	2017-03-05 11:05:24.000000000 -0600
+++ src/gnutls-3.3.27/tests/Makefile.am	2017-05-01 22:15:39.777189100 -0500
@@ -53,9 +53,10 @@ AM_CPPFLAGS = \
 	$(P11_KIT_CFLAGS)
 
 AM_LDFLAGS = -no-install
-LDADD = ../lib/libgnutls.la \
+LDADD = \
 	../gl/libgnu.la \
 	libutils.la \
+	../lib/libgnutls.la \
 	$(LTLIBGCRYPT)	\
 	$(LIBSOCKET) $(INET_NTOP_LIB) $(INET_PTON_LIB)
 
