--- origsrc/gnutls-2.4.1/configure.in	2008-06-30 15:45:51.000000000 -0500
+++ src/gnutls-2.4.1/configure.in	2008-07-17 14:21:50.984375000 -0500
@@ -102,7 +102,7 @@
 AC_C_INLINE
 
 AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.17])
+AM_GNU_GETTEXT_VERSION([0.15])
 
 AC_ARG_ENABLE(cxx,
        AS_HELP_STRING([--disable-cxx],
@@ -608,6 +608,8 @@
 
 # Windows hacks.
 
+case "$host_os" in
+mingw*)
 AC_DEFINE([WINVER], [0x0501], [Windows 2000 or before not supported.])
 save_LIBS="$LIBS"
 LIBS="$LIBS -lws2_32"
@@ -617,6 +619,9 @@
   int err = WSAStartup(wVersionRequested, &wsaData);
   WSACleanup ();], wsastartup=yes, wsastartup=no)
 LIBS="$save_LIBS"
+;;
+*)	wsastartup=no ;;
+esac
 if test "$wsastartup" = "yes"; then
   AC_DEFINE(HAVE_WINSOCK, 1, [Call WSAStartup in gnutls_global_init])
 fi
--- origsrc/gnutls-2.4.1/guile/src/Makefile.am	2008-06-19 05:00:10.000000000 -0500
+++ src/gnutls-2.4.1/guile/src/Makefile.am	2008-07-17 01:19:29.328125000 -0500
@@ -46,6 +46,7 @@
 libguile_gnutls_v_1_la_SOURCES = core.c errors.c utils.c
 libguile_gnutls_v_1_la_CFLAGS =			\
   $(AM_CFLAGS) $(GNULIB_CFLAGS) $(GUILE_CFLAGS)
+libguile_gnutls_v_1_la_LDFLAGS = -module -avoid-version -no-undefined
 libguile_gnutls_v_1_la_LIBADD = \
 	$(GNUTLS_CORE_LIBS) $(GNULIB_LIBS) \
 	$(GUILE_LDFLAGS)
@@ -53,6 +54,7 @@
 libguile_gnutls_extra_v_1_la_SOURCES = extra.c
 libguile_gnutls_extra_v_1_la_CFLAGS =		\
   $(AM_CFLAGS) $(GNULIB_CFLAGS) $(GUILE_CFLAGS)
+libguile_gnutls_extra_v_1_la_LDFLAGS = -module -avoid-version -no-undefined
 libguile_gnutls_extra_v_1_la_LIBADD =		\
   $(GNUTLS_CORE_LIBS) $(GNUTLS_EXTRA_LIBS)	\
   $(builddir)/libguile-gnutls-v-1.la		\
--- origsrc/gnutls-2.4.1/guile/src/utils.h	2008-06-19 05:00:10.000000000 -0500
+++ src/gnutls-2.4.1/guile/src/utils.h	2008-07-17 01:19:29.359375000 -0500
@@ -43,7 +43,7 @@
 
 /* Arrays as byte vectors.  */
 
-extern const char scm_gnutls_array_error_message[];
+SCM_API const char scm_gnutls_array_error_message[];
 
 /* Initialize C_HANDLE and C_LEN and return the contiguous C array
    corresponding to ARRAY.  */
--- origsrc/gnutls-2.4.1/lib/Makefile.am	2008-06-30 15:45:51.000000000 -0500
+++ src/gnutls-2.4.1/lib/Makefile.am	2008-07-17 01:19:29.390625000 -0500
@@ -143,6 +143,7 @@
 defexecdir = $(bindir)
 defexec_DATA = libgnutls-$(SOVERSION).def
 DISTCLEANFILES += $(defexec_DATA)
+libgnutls-$(SOVERSION).def: libgnutls.la ;
 endif
 
 # C++ library
--- origsrc/gnutls-2.4.1/libextra/Makefile.am	2008-06-19 05:00:10.000000000 -0500
+++ src/gnutls-2.4.1/libextra/Makefile.am	2008-07-17 01:19:29.421875000 -0500
@@ -76,6 +76,7 @@
 libgnutls_openssl_la_LDFLAGS += \
 	-Wl,--output-def,libgnutls-openssl-$(SOVERSION).def
 defexec_DATA += libgnutls-openssl-$(SOVERSION).def
+libgnutls-openssl-$(SOVERSION).def: libgnutls-openssl.la ;
 endif
 endif
 
@@ -94,6 +95,7 @@
 libgnutls_extra_la_LDFLAGS += \
 	-Wl,--output-def,libgnutls-extra-$(SOVERSION).def
 defexec_DATA += libgnutls-extra-$(SOVERSION).def
+libgnutls-extra-$(SOVERSION).def: libgnutls-extra.la ;
 endif
 
 DISTCLEANFILES += $(defexec_DATA)
--- origsrc/gnutls-2.4.1/src/cli.c	2008-06-19 05:35:13.000000000 -0500
+++ src/gnutls-2.4.1/src/cli.c	2008-07-17 15:05:00.968750000 -0500
@@ -40,6 +40,7 @@
 #include "error.h"
 #include "read-file.h"
 #include "getpass.h"
+#include "getaddrinfo.h"
 
 #include "common.h"
 #include "cli-gaa.h"
--- origsrc/gnutls-2.4.1/src/serv.c	2008-06-19 05:00:10.000000000 -0500
+++ src/gnutls-2.4.1/src/serv.c	2008-07-17 15:03:54.843750000 -0500
@@ -778,8 +778,10 @@
 {
   switch (addr->ss_family)
     {
+#ifdef HAVE_IPV6
       case AF_INET6:
         return ntohs (((const struct sockaddr_in6 *)addr)->sin6_port);
+#endif
       case AF_INET:
         return ntohs (((const struct sockaddr_in *)addr)->sin_port);
     }
--- origsrc/gnutls-2.4.1/src/tls_test.c	2008-06-19 05:00:10.000000000 -0500
+++ src/gnutls-2.4.1/src/tls_test.c	2008-07-17 15:06:16.062500000 -0500
@@ -31,6 +31,7 @@
 #include <tests.h>
 #include <common.h>
 #include <tls_test-gaa.h>
+#include "getaddrinfo.h"
 
 #define SA struct sockaddr
 #define ERR(err,s) if (err==-1) {perror(s);return(1);}
